src/NumericalAlgorithms/RootFinding/TOMS748.hpp

```diff
@@ -68,6 +68,11 @@ double toms748(const Function& f, const double lower_bound,
  *
  * \snippet Test_TOMS748.cpp datavector_root_find
  *
+ * The upper and lower bounds must bracket the root so that
```

Comment:
Why not just say that the root find searches in (a, b) not [a, b]?

Reply:
I went looking in boost... their documentation says

> It is a precondition that ... `f(a) * f(b) < 0`

but their source code reads `if ((fa == 0) || (fb == 0)) { return one of a or b }`, which would be better matched by `<=` in the documentation.

I suggest saying, based on your comment but making it consistent with the boost code,

"The TOMS748 algorithm searches for a root in the interval [`lower_bound`, `upper_bound`], and will throw an exception if this interval does not bracket a root, i.e. if `f(lower_bound) * f(upper_bound) > 0`.